sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -244,28 +245,31 @@ class NodeHttpClient implements HttpClient {
   }
 
   private getOrCreateAgent(request: PipelineRequest, isInsecure: boolean): http.Agent {
-    if (!request.disableKeepAlive) {
-      if (isInsecure) {
-        if (!this.httpKeepAliveAgent) {
-          this.httpKeepAliveAgent = new http.Agent({
-            keepAlive: true,
-          });
-        }
+    if (!request.tlsSettings && request.disableKeepAlive) {
+      return isInsecure ? http.globalAgent : https.globalAgent;
+    }
+    const settingsIndex = request.tlsSettings ?? {};
+    let agent: http.Agent | undefined = isInsecure
+      ? this.httpKeepAliveAgent
+      : this.httpsKeepAliveAgent?.get(settingsIndex);
 
-        return this.httpKeepAliveAgent;
-      } else {
-        if (!this.httpsKeepAliveAgent) {
-          this.httpsKeepAliveAgent = new https.Agent({
-            keepAlive: true,
-          });
-        }
+    if (agent) {
+      return agent;
+    }
 
-        return this.httpsKeepAliveAgent;
-      }
-    } else if (isInsecure) {
-      return http.globalAgent;
+    const agentOptions: https.AgentOptions = {
+      ...request.tlsSettings,
+      keepAlive: !request.disableKeepAlive,
```

Comment:
since we're indexing only on the tlsSettings, won't this potentially create a scenario where if request 1 has `disableKeepAlive: true`, but then request2 has the *same* tlsSettings with `disableKeepAlive: false` we will use the cached agent which has keep alive turned off?